mesonbuild/compilers/compilers.py

```diff
@@ -1561,6 +1561,9 @@ def openmp_flags(self):
         else:
             return ['-openmp']
 
+    def get_link_whole_for(self, args):
+        return ['-Wl,--whole-archive'] + args + ['-Wl,--no-whole-archive']
```

Comment:
I was actually thinking that this is a property of the linker, so the implementation will always be the same as the GnuCompiler. It makes sense to just use the GnuCompiler implementation in that case, like:

```
GnuCompiler.get_link_whole_for(self, args)
```

Reply:
Ok I'll update